components/net/fetch/methods.rs

```diff
@@ -969,3 +971,27 @@ fn should_upgrade_request_to_potentially_trustworty(
     // Step 4
     request.insecure_requests_policy == InsecureRequestsPolicy::Upgrade
 }
+
+/// <https://w3c.github.io/webappsec-mixed-content/#upgrade-algorithm>
+fn should_upgrade_mixed_content_request(request: &Request) -> bool {
+    let url = request.url();
+
+    // Step 1.
+    //TODO: handle checking for environment setting
```

Comment:
I believe this TODO needs to be handled to avoid all of the timeouts in the tests.